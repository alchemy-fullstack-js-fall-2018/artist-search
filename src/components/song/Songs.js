import React, { Component, Fragment } from 'react';
import { getLyrics } from '../../services/lyricSearch';

export default class Song extends Component {

  state = {
    lyrics: '',
    artist: '',
    song: ''
  };

  componentDidMount() {
    const artist = this.props.location.pathname.slice(1).split('/')[1];
    const song = this.props.location.pathname.slice(1).split('/')[2];

    getLyrics(artist, song)
      .then(result => {
        this.setState({ lyrics: result.lyrics, artist, song, });
      });
  }

  render() {
    const { artist, song, lyrics } = this.state;

    return (
      <Fragment>
        <h3>{artist}</h3>
        <h4>{song}</h4>
        <p>{lyrics}</p>
      </Fragment>
    );
  }
}
