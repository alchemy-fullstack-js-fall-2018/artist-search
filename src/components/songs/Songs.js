import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import PropTypes from 'prop-types';

const Song = ({ artist, title }) => {
  return (
    <Fragment>
      <h4><em>{title}</em></h4>
      <Link to={ROUTES.LYRICS.linkTo(artist, title)}>Get Lyrics</Link>
    </Fragment>
  );
};

function Songs({ artist, works }) {
  return (
    <Fragment>
      <h3>Songs:</h3>
      {works.map(work =>
        <Song
          key={work.id}
          artist={artist}
          title={work.title}
          id={work.id}
        />)
      }
    </Fragment>
  );
}

Songs.propTypes = {
  artist: PropTypes.string.isRequired,
  works: PropTypes.array.isRequired
};

export default Songs;
