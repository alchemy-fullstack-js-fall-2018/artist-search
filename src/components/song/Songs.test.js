import React from 'react';
import Songs from './Songs';
import { shallow } from 'enzyme';

jest.mock('../../services/lyricSearch', () => {
  return { getLyrics: () => Promise.resolve({ lyrics: 'blah' }) };
});

describe('Songs', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Songs location={{ pathname: '/somepath' }}/>);
  });
  it('renders a Songs component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays artist, song and lyrics', () => {
    expect(wrapper.html()).toContain('blah');
  });
});
