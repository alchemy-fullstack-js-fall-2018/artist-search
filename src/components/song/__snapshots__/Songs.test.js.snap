// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs renders a Songs component 1`] = `
<Fragment>
  <h3 />
  <h4 />
  <p>
    blah
  </p>
</Fragment>
`;
